### PR TITLE
Add lesson about Enumerator#size

### PR DESCRIPTION
**Pull request by omartell**

### CODE REVIEW

File: ruby/enumerator_size.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  result_from_api.each do |n|
+    yielder << n
+  end
+end
+```
+
+Then calling `#size` gives us a more useful value:
+
+```ruby
+wraper.size => 3
```
Comment (scottmatthewman) [inline ruby/enumerator_size.md]: Typo - should be `wrapper`

File: ruby/enumerator_size.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  result_from_api.each do |n|
+    yielder << n
+  end
+end
+```
+
+Then, if you call `#size`, you'd get an infamous `nil`:
+
+```ruby
+  wrapper.size # => nil
```
Comment (scottmatthewman) [inline ruby/enumerator_size.md]: For consistency's sake, we should probably remove the unnecessary indentation on this line.

File: ruby/enumerator_size.md (hunk 1 above)
Comment (omartell) [inline ruby/enumerator_size.md]: good catch

File: ruby/enumerator_size.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+result_from_api = [1, 2, 3]
+
+wrapper = Enumerator.new do |yielder|
+  result_from_api.each do |n|
+    yielder << n
+  end
+end
+```
+
+Then, if you call `#size`, you'd get an infamous `nil`:
```
Comment (scottmatthewman) [inline ruby/enumerator_size.md]: Do we actually mean "infamous" here?

Comment (omartell) [inline ruby/enumerator_size.md]: You mean the word "infamous" is not appropriate in this context?

Comment (scottmatthewman) [inline ruby/enumerator_size.md]: It doesn't feel like it is.

Comment (omartell) [inline ruby/enumerator_size.md]: Ok, yeah, wasn't completely sure.

File: ruby/enumerator_size.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+```ruby
+  wrapper.size # => nil
+```
+
+And that's correct as Ruby does not know if the enumeration has
+actually finshed. However, after looking at the documentation we noticed
```
Comment (jbilbo) [inline ruby/enumerator_size.md]: typo `finished`

Comment (omartell) [inline ruby/enumerator_size.md]: @jbilbo :+1:

**Comment by omartell** [comment]
@jbilbo @scottmatthewman updated

**Pull request merged**